### PR TITLE
Userpage 

### LINKED ISSUE #176: User page
Have a page where a user can see all their assignments and their latest submissions.

### PR DESCRIPTION
**Pull request by BiancaNecula**
Fixes #176 

Create a page where the user can see all their submissions to assignments.

### CODE REVIEW

File: interface/templates/interface/generics/navbar.html [hunk 1]
```diff
@@ -20,10 +20,12 @@
             <a class="nav-link" href="{%  url 'upload' assignment.course.pk assignment.pk %}">Upload</a>
           </li>
         {% endif %}
-
-          <li class="nav-item">
-                <a class="nav-link" href="{% url 'logout' %}">Logout</a>
-          </li>
+        <li class="nav-item">
+          <a class="nav-link" href="{% url 'user_page' user.username %}">My submissions</a>
```
Comment (gmuraru) [inline interface/templates/interface/generics/navbar.html:24]: PICKY: Could you indent this as the one from line 27?

**Comment by gmuraru** [comment]
Could you also update the documentation, by specifying the ```UserPage```? (you can find it at the top of the page in the ```Wiki``` section)

File: interface/templates/interface/user_page.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{% extends "interface/generics/base.html" %}
+
+{% block body_content %}
+  <div class="container">
+    <h1> Submisii {{ user.username }} </h1>
```
Comment (Jokeswar) [inline interface/templates/interface/user_page.html:5]: Let's stick to english

File: interface/views.py [hunk 3]
```diff
@@ -296,3 +296,16 @@ def subs_for_user(request, course_pk, assignment_pk, username):
...
+
+
+@login_required
+def user_page(request, username):
+    user = get_object_or_404(User, username=username)
+    submissions_aux = Submission.objects.all().order_by('-timestamp')
+    submissions = submissions_aux.filter(user=user)
+    if request.user.username != username:
+        log.warning(f'User attempted to access {username}')
+        raise Http404("You are not allowed to access this page.")
```
Comment (Jokeswar) [inline interface/views.py:308]: I think we should have this `if` after line 303. There is no need for a submission querry if the user is not valid

File: interface/views.py [hunk 4]
```diff
@@ -296,3 +296,16 @@ def subs_for_user(request, course_pk, assignment_pk, username):
         'assignment': assignment,
         'submissions': submissions,
     })
+
+
+@login_required
+def user_page(request, username):
+    user = get_object_or_404(User, username=username)
+    submissions_aux = Submission.objects.all().order_by('-timestamp')
+    submissions = submissions_aux.filter(user=user)
```
Comment (Jokeswar) [inline interface/views.py:305]: This can be a single line. (e.g: `Submission.objects.all().filter(user=user).order_by('-timestamp')`)

**Comment by BiancaNecula** [comment]
userpage

File: testsuite/test_student_workflow.py [hunk 5]
```diff
@@ -198,3 +198,17 @@ def test_anonymous(client, STC):
 
     response = client.get('/submission/')
     STC.assertRedirects(response, '/?next=/submission/')
+
+
+@pytest.mark.django_db
+def test_a(client, base_db_setup):
```
Comment (Jokeswar) [inline testsuite/test_student_workflow.py:204]: I missed it, my bad! Let's give it a proper name.

**Pull request merged**